### DISCUSSION TITLE
proc macro * not expanded: proc macro crate is missing dylib

### DISCUSSION THREAD

**Discussion by Jacherr** [rust-lang/rust-analyzer#17342] [Q&A] [answered]
Not sure if this is better suited as a bug report or not, but recently I cloned https://github.com/rust-lang/rust and configured rust-analyzer for development on rustc. Ever since then, however, other unrelated projects have begun to show this error on all proc macros. I'm not really sure what setting may have been messed with to break this, but I can't really find much about it online.

I'm on the latest Nightly (`rustc 1.80.0-nightly (7c52d2db6 2024-06-03)`), with Visual Studio Code. VSC and rust-analyzer are both the latest (stable) versions. Any help appreciated!

Edit: My initial thinking is it's something to do with a proc macro server configuration conflict or something. All rustc-specific configuration does seem to be confined to that project's `setting.json` so not sure how that could impact anything?

**Comment by Veykril** [comment, marked as answer]
Hmm, the only thiings that can affect it here are modifying your editors `settings.json` (double check that file) or having somehow set a toolchain override that might mess with things.

**Reply by Jacherr** [reply]
Turns out that after a bit of further investigation my global `settings.json` had somehow been overwritten by one from another project and it was causing all kinds of odd behaviour, thanks for the help!

**Comment by i4han** [comment]
I had the same issue after `rm -rf`d one of rust project in a directory. 
The issue was solved after I edited `settings.json` "rust-analyzer.linkedProjects": [...]. Removed the directory from the array.
